pdpserver/server/log_mgr.go

```diff
@@ -0,0 +1,19 @@
+package server
+
+import (
+	"github.com/Sirupsen/logrus"
+)
+
+type LevelCheckable interface {
+	GetLevel() logrus.Level
+}
+
+func GetLogLevel(logger logrus.FieldLogger) logrus.Level {
+	if lgr, ok := logger.(*logrus.Logger); ok {
```

Comment:
Everywhere when you need to distinct more than one type it's simpler to use type switch. 
```go
switch log := logger.(type) {
case *logrus.Logger:
	return log.Level

case LevelCheckable:
	return log.GetLevel()
}
```